src/loadScriptFile.ts

```diff
@@ -11,34 +11,39 @@ import { PackageJson } from './parsers/parsePackageJson';
 import LogCategory = rpc.RpcLog.RpcLogCategory;
 import LogLevel = rpc.RpcLog.Level;
 
+let hasLoggedAttempt = 0;
+let hasLoggedWarning = false;
+
 export async function loadScriptFile(filePath: string, packageJson: PackageJson): Promise<unknown> {
     // See the following issue for more details on why we want to retry
     // https://github.com/Azure/azure-functions-nodejs-worker/issues/693
-    const fileName = path.basename(filePath);
+    const retries = 9;
     return await retry(
         async (currentAttempt: number) => {
-            if (currentAttempt > 1) {
+            if (currentAttempt > 1 && currentAttempt > hasLoggedAttempt) {
                 worker.log({
-                    message: `Retrying load of file "${fileName}". Attempt ${currentAttempt}/${10}`,
+                    message: `Retrying file load. Attempt ${currentAttempt}/${retries + 1}`,
                     level: LogLevel.Debug,
                     logCategory: LogCategory.System,
                 });
+                hasLoggedAttempt = currentAttempt;
             }
             return loadScriptFileInternal(filePath, packageJson);
         },
         {
-            retries: 9,
-            minTimeout: 50,
+            retries: retries,
+            minTimeout: 500,
             onFailedAttempt: (error) => {
                 if (!/lstat.*home/i.test(error?.message || '')) {
                     // this will abort the retries if it's an error we don't recognize
                     throw error;
-                } else if (error.retriesLeft > 0) {
+                } else if (error.retriesLeft > 0 && !hasLoggedWarning) {
                     worker.log({
-                        message: `Warning: Failed to load file "${fileName}" with error "${error.message}"`,
+                        message: `Warning: Failed to load file with error "${error.message}"`,
```

Comment:
I do think it's helpful to know which file exactly failed to load, at least on the initial attempt

Reply:
Right, I see now that this is only for the lstat error case, and in other errors, we throw the error. In those cases, would the erroneous file name still be logged somewhere?

Reply:
For app start we log the file name yes:

https://github.com/Azure/azure-functions-nodejs-worker/blob/f3f6219d157b8119f62d75bd948e95775bbf8653/src/startApp.ts#L102

For function load (this only applies to the old model), seems like we only log function name, but my guess is that's usually enough idk:

https://github.com/Azure/azure-functions-nodejs-worker/blob/f3f6219d157b8119f62d75bd948e95775bbf8653/src/eventHandlers/FunctionLoadHandler.ts#L46

Reply:
Got it, makes sense